core/geometry/src/polyface/Polyface.ts

```diff
@@ -239,19 +236,19 @@ export class IndexedPolyface extends Polyface {
     }
 
     // Add normal and normal index data
-    if (copyNormals && source.data.normal) {
-      const startOfNewNormals = this.data.normal!.length;
-      const numNewNormals = source.data.normal.length;
+    if (copyNormals) {
+      const myNormals = this.data.normal!;
```

Comment:
Can you replace ! with assert here?

Reply:
copyNormals won't be true if this.data.normal is undefined thanks to the call to allDefined() in line 189.  Should I still assert?

Reply:
Yes, that's exactly what `assert` is for - to tell the compiler the truth of something it can't determine on its own.

Reply:
I cleaned up the code and removed a bunch of bangs. This should remove the need for the asserts. But I'll keep `assert` in mind from now on. Thanks for the input!